config-overrides.js

```diff
@@ -1,13 +1,24 @@
 const webpack = require('webpack');
+const {
+  useBabelRc,
+  override,
+  useEslintRc,
+  addBabelPreset,
+} = require('customize-cra');
 
 module.exports = {
-  webpack: function (config, env) {
-    config.plugins.push(
-      new webpack.ProvidePlugin({
-        L: 'leaflet',
-      })
-    );
+  webpack: override(
+    useBabelRc(),
+    useEslintRc(),
+    addBabelPreset('@emotion/babel-preset-css-prop'),
+    function (config, env) {
+      config.plugins.push(
+        new webpack.ProvidePlugin({
+          L: 'leaflet',
+        })
+      );
 
-    return config;
-  },
+      return config;
+    }
+  ),
 };
```

Comment:
I wonder if we should do away with react-rewired- app entirely to reduce the number of dependencies and deviation from a standard create react app application.

We could use the:
- JSX pragma when we need to use the css prop (unless styled-system relies on it?)
- It looks like with import leaflet everywhere it's needed so we don't need the ProvidePlugin for that

What do you think @falconx?

Reply:
I'm hoping it's not necessary to use the CSS prop, so I almost didn't add it.

We could use the JSX pragma and it's not a requirement for styled-system however it does mean their own [css prop documentation](https://styled-system.com/css) would only half work too — in this example, without `@emotion/babel-preset-css-prop`, the `Boop` component would be valid, but `Beep` would not.

So whilst I imagine we could get away without it, given that it looks like the config overrides are here to say anyway, it might be worth keeping? I'm also thinking, even if you and I don't make use of it (and I haven't used either enough to know this likelihood), it could be a pain if anyone comes onto the project and expects those features to work — which would save us documenting that they don't. That said, I'm happy to go with a philosophy of removing and only adding when/if we find it necessary :)

Reply:
yup agreed, let's keep it in 👍